waiter/integration/waiter/kubernetes_scheduler_integration_test.clj

```diff
@@ -337,3 +337,36 @@
                     :interval 15
                     :timeout timeout-secs)))))
         (log/warn "skipping test as INTEGRATION_TEST_BAD_IMAGE is not specified")))))
+
+(deftest ^:parallel ^:integration-fast test-kubernetes-reverse-proxy-sidecar-headers
+  (testing-using-waiter-url
+    (when (using-k8s? waiter-url)
+      (let [{:keys [service-id] :as response}
+            (make-request-with-debug-info
+              {:x-waiter-distribution-scheme "simple"
+               :x-waiter-name (rand-name)
+               :x-waiter-env-GRPC_TRANSCODER "yes"}
+              #(make-kitchen-request waiter-url % :method :get :path "/request-info"))]
+        (println (with-out-str (clojure.pprint/pprint response)))
+        (with-service-cleanup
+          service-id
+          (assert-response-status response http-200-ok)
+          (testing "The envoy x-envoy-expected-rq-timeout-ms header is present"
+            (is (contains? (:headers response) "x-envoy-expected-rq-timeout-ms"))))))))
+
+(deftest ^:parallel ^:integration-fast test-kubernetes-reverse-proxy-sidecar-port
+  (testing-using-waiter-url
+    (when (using-k8s? waiter-url)
+      (let [{:keys [service-id] :as response}
+            (make-request-with-debug-info
+              {:x-waiter-distribution-scheme "simple"
+               :x-waiter-name (rand-name)
+               :x-waiter-env-GRPC_TRANSCODER "yes"}
+              #(make-kitchen-request waiter-url % :method :get :path "/environment"))]
+        (println (with-out-str (clojure.pprint/pprint response)))
+        (with-service-cleanup
+          service-id
+          (assert-response-status response http-200-ok)
+          (testing "Port value is correctly offset compared to instance value"
+            (is (= (+ 1 (Integer/parseInt (get (:headers response) "x-waiter-backend-port")))
+                   (Integer/parseInt (get (:body response) "PORT0"))))))))))
```

Comment:
You need to parse the body as JSON, otherwise it's just a string. Use the `try-parse-json` utility function (already pulled into this namespace from `waiter.util.client-tools`).

Reply:
Also, this would be more readable if you used a `let` to create local bindings for `backend-port` and `port0`.
Clojure's threading macro is also nice for this type of thing:

```
(let [backend-port (-> response :headers (get "x-waiter-backend-port") Integer/parseInt)
      ...
```

Reply:
Done